solr/core/src/java/org/apache/solr/handler/component/QueryComponent.java

```diff
@@ -910,8 +910,12 @@ protected void createMainQuery(ResponseBuilder rb) {
         additionalAdded = addFL(additionalFL, "score", additionalAdded);
       }
     } else {
-      // reset so that only unique key is requested in shard requests
-      sreq.params.set(CommonParams.FL, rb.req.getSchema().getUniqueKeyField().getName());
+      if (rb.req.getSearcher().enableLazyFieldLoading) {
+        // reset so that only unique key is requested in shard requests
+        sreq.params.set(CommonParams.FL, rb.req.getSchema().getUniqueKeyField().getName());
+      } else {
+        sreq.params.set(CommonParams.FL, "*");
```

Comment:
I can see why this change was made but my point is that there is probably an assumption in the join queries or in the test (I haven't looked) which make it fail without this change.